File: collatz.c
# include <stdio.h>
# include <time.h>

int collatz(int);

int main(){
    
    int clocktime;
    clocktime = clock();
    int n;
    FILE *fp;
    fp = fopen("collatz.dat", "w");

    for (n = 1; n < 100001; n++)
    {
        fprintf(fp, "%d %d\n", n, collatz(n));
    }
    
    fclose(fp);
    printf("%ld\n", (clock() - clocktime));
    return 0;
    
}

int collatz(int x){
    
    int i;
    int count;
    count = 0;
    while (i != 1)
    {
        count += 1;
        if (i % 2 == 0) i /= 2;
        else i = 3 * i + 1;
    }
    return count;

}

File: collatz.py
import time

t0 = time.time()

def collatz(n):
    count = 0
    while n != 1:
        count += 1
        if n % 2 == 0:
            n /= 2
        else: n = 3 * n + 1
    return count

with open("collatz_py.dat", "w") as fp:
    for n in range (1, 100_001):
        fp.write(str(n)+" "+str(collatz(n))+"\n")

print(time.time() - t0)